internal/pkg/service/buffer/store/model/mapper/mapper_test.go

```diff
@@ -0,0 +1,207 @@
+  "secret": "%s",
+  "Exports": [
+    {
+      "projectId": 1000,
+      "receiverId": "receiver",
+      "exportId": "export",
+      "name": "Export",
+      "importConditions": {
+        "count": 1000,
+        "size": "100MB",
+        "time": 180000000000
+      },
+      "revisionId": 0,
+      "tableId": {
+        "stage": "in",
+        "bucketName": "bucket",
+        "tableName": "table"
+      },
+      "incremental": false,
+      "columns": [
+        {
+          "type": "body"
+        },
+        {
+          "type": "template",
+          "language": "jsonnet",
+          "undefinedValueStrategy": "null",
+          "content": "a+\":\"+b",
+          "dataType": "STRING"
+        }
+      ]
+    }
+  ]
+}
+`,
+		json.MustEncodeString(model, true),
+	)
+}
+
+func TestMapperReceiverModelToPayload(t *testing.T) {
+	t.Parallel()
+
+	receiverKey := key.ReceiverKey{
+		ProjectID:  1000,
+		ReceiverID: "receiver",
+	}
+	exportKey := key.ExportKey{
+		ReceiverKey: receiverKey,
+		ExportID:    "export",
+	}
+	mappingKey := key.MappingKey{
+		ExportKey:  exportKey,
+		RevisionID: 1,
+	}
+
+	model := model.Receiver{
+		ReceiverBase: model.ReceiverBase{
+			ReceiverKey: receiverKey,
+			Name:        "Receiver",
+			Secret:      "test",
+		},
+		Exports: []model.Export{
+			{
+				ExportBase: model.ExportBase{
+					ExportKey: exportKey,
+					Name:      "Export",
+					ImportConditions: model.ImportConditions{
+						Count: 1000,
+						Size:  100,
+						Time:  100_000_000_000,
+					},
+				},
+				Mapping: model.Mapping{
+					MappingKey: mappingKey,
+					TableID: model.TableID{
+						Stage:  model.TableStageIn,
+						Bucket: "bucket",
+						Table:  "table",
+					},
+					Incremental: false,
+					Columns: column.Columns{
+						column.Body{},
+						column.Template{
+							Language:               "jsonnet",
+							UndefinedValueStrategy: "null",
+							Content:                `a+":"+b`,
+							DataType:               "STRING",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	payload := ReceiverPayloadFromModel("buffer.keboola.local", model)
+	assert.Equal(t,
+		`{
+  "ID": "receiver",
+  "URL": "https://buffer.keboola.local/v1/import/1000/receiver/test",
+  "Name": "Receiver",
+  "Exports": [
+    {
+      "ID": "export",
+      "ReceiverID": "receiver",
+      "Name": "Export",
+      "Mapping": {
+        "TableID": "in.c-bucket.table",
+        "Incremental": false,
+        "Columns": [
+          {
+            "Type": "body",
+            "Template": null
+          },
+          {
+            "Type": "template",
+            "Template": {
+              "Language": "jsonnet",
+              "UndefinedValueStrategy": "null",
+              "Content": "a+\":\"+b",
+              "DataType": "STRING"
+            }
+          }
+        ]
+      },
+      "Conditions": {
+        "Count": 1000,
+        "Size": "100B",
+        "Time": "1m40s"
+      }
+    }
+  ]
+}
+`,
+		json.MustEncodeString(payload, true),
```

Comment:
Payload by si myslim nemal porovnavat ako string ale skor ako strukturu (da sa to jednoducho zapisat v podobnom tvare), ... teoreticky aj model, ale tam to az tak nevadi. Payload nema definovane JSON tags, kedze si to API riesi na inej urovni dalej.

Ako prezijem aj s tymto, rozhodni sa sam.

Reply:
Tam je problém v tom, že se pak hůř dělá comparison s tím `secret`, který je random generated. Existuje nějaký `wildcards.Equal`, kterým pak jde ignorovat ten secret? Nebo mám dát `ReceiverModelFromPayload` a pak vymazat z té struktury ten secret? Nevím, tohle mi přišlo jednodušší a v podstatě je to asi jedno 😄

Reply:
No ale mapper negeneruje nahodne hodnoty nie? (to nie je jeho uloha) Tu by to nemal byt problem.

Vo vseobecnosti by som nahodnu hodnotu odstranil, priamo, alebo cez nejaky for, ak je to slice.
Tj. skontroloval by som ze nie je prazdna, ma spravnu dlzku, odstranil by som ju.

A potom by som porovnal zvysok cez `assert.Equal`.